Sources/StreamChatSwiftUI/ChatChannel/Composer/MessageComposerViewModel.swift

```diff
@@ -455,4 +464,35 @@ open class MessageComposerViewModel: ObservableObject {
                 .store(in: &cancellables)
         }
     }
+    
+    private func listenToCooldownUpdates() {
+        channelController.channelChangePublisher.sink { [weak self] _ in
+            let cooldownDuration = self?.channelController.channel?.cooldownDuration ?? 0
+            if self?.cooldownPeriod == cooldownDuration {
+                return
+            }
+            self?.cooldownPeriod = cooldownDuration
+            self?.checkChannelCooldown()
+        }
+        .store(in: &cancellables)
+    }
```

Comment:
This is just an idea, and I'm not completely sure it's better, but we COULD make this a little more combine-y with something along the lines of (pseudo-code, not working):

```suggestion
    private func listenToCooldownUpdates() {
        channelController.channelChangePublisher
        .map { self?.channelController.channel?.cooldownDuration ?? 0 }
        .filter { self?.cooldownPeriod != $0 }
        .sink { [weak self] _ in
            self?.cooldownPeriod = cooldownDuration
            self?.checkChannelCooldown()
        }
        .store(in: &cancellables)
    }
```

Reply:
As discussed, we will keep the other implementation for readability.